dubbo-metrics/dubbo-metrics-prometheus/src/main/java/org/apache/dubbo/metrics/prometheus/PrometheusMetricsReporter.java

```diff
@@ -93,7 +94,8 @@ private void exportHttpServer() {
                     }
                 });
 
-                new Thread(prometheusExporterHttpServer::start).start();
+                httpServerThread = new Thread(prometheusExporterHttpServer::start);
+                httpServerThread.start();
```

Comment:
The http server management seems a bit strange, see about: https://stackoverflow.com/a/2917297
```java
    this.httpServer = HttpServer.create(addr, 0);
    HttpContext context = this.httpServer.createContext("/", new DocumentProcessHandler());
    this.httpThreadPool = Executors.newFixedThreadPool(this.noOfThreads);
    this.httpServer.setExecutor(this.httpThreadPool);
    this.httpServer.start();
```

```java
        this.httpServer.stop(1);
        this.httpThreadPool.shutdownNow();
```

Reply:
The `httpServerThread` is terminated after stop http server.